src/Compilers/Core/Portable/Compilation/RebuildData.cs

```diff
@@ -10,7 +10,7 @@ namespace Microsoft.CodeAnalysis
 {
     internal sealed class RebuildData
     {
-        private Queue<string> NonSourceFileDocumentNames { get; } = new();
+        private Queue<string> NonSourceFileDocumentNames { get; }
```

Comment:
Might be worth calling out `#line` directives as a source of these document names.